PayloadPkg/OsLoader/LoadImage.c

```diff
@@ -299,13 +328,40 @@ LoadLinuxFile (
   AsciiStrToUnicodeStrS (Ptr, FileName, sizeof(FileName) / sizeof(CHAR16));
   FileSize   = 0;
   FileBuffer = NULL;
-  Status = GetFileByName (FsHandle, FileName, &FileBuffer, &FileSize);
-  DEBUG ((DEBUG_INFO, "Load file %a [size 0x%x]: %r\n", Ptr, FileSize, Status));
+  FileHandle = NULL;
+  Status = OpenFile (FsHandle, FileName, &FileHandle);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "Open file '%s' failed, Status = %r\n", FileName, Status));
+    goto Done;
+  }
+
+  Status = GetFileSize (FileHandle, &FileSize);
+  if (EFI_ERROR (Status)) {
+    DEBUG ((DEBUG_INFO, "Get file '%s' size failed, Status = %r\n", FileName, Status));
+    goto Done;
+  }
+
+  FileBuffer = AllocatePool (FileSize);
+  if (FileBuffer == NULL) {
+    Status = EFI_OUT_OF_RESOURCES;
+    goto Done;
+  }
+
+  Status = ReadFile (FileHandle, &FileBuffer, &FileSize);

```

Comment:
is it possible to make the APIs exactly same for FileSystemLib and specific file system (e.g. Fat)? look currently FsHandle is not exposed in FileSystemLib.

Reply:
This is known issue. Currently, FatReadFile requires FsHandle as a parameter. Let me change FatReadFile in separate commit.